### PR TITLE
improvement resource handlers

### PR DESCRIPTION
**Pull request by decebals**
Attempt to solve the drawbacks found via #170. It's only a suggestion, any comments are welcome.

### CODE REVIEW

**Comment by gitblit** [comment]
This looks good. I haven't run it yet. Instead of doing Application.get() in handle() you've got routeContext.getApplication ()

File: pippo-core/src/main/java/ro/pippo/core/route/UrlResourceHandler.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        if (url == null) {
+            routeContext.getResponse().notFound().commit();
+        } else {
+            streamResource(url, routeContext);
+        }
+    }
+
+    public abstract URL getResourceUrl(String resourcePath);
+
+    protected void streamResource(URL resourceUrl, RouteContext routeContext) {
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/route/UrlResourceHandler.java]: I can remove this method and put the content inline directly in the method `handleResource`.  I don't know if someone wants to bypass this part with caching.

File: pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java [hunk 2]
```diff
@@ -39,37 +36,30 @@
...
-        URL url = super.getResourceUrl(resourcePath);
-        if (url == null) {
-            return null;
-        }
-
+    protected void sendResource(URL resourceUrl, RouteContext routeContext) throws IOException {
         // clear cache for DEV mode
-        if (Application.get().getRuntimeMode().equals(RuntimeMode.DEV)) {
+        if (routeContext.getApplication().getPippoSettings().isDev()) {
             sourceMap.clear();
```
Comment (decebals) [inline pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java]: Is it better to touch the cache (`sourceMap`) only when I am  in `PROD` mode (no `sourceMap.get()`, `sourceMap.put()` in this mode)?

Comment (danjee) [inline pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java]: I agree with that. So to put records on that map only if in production mode. Do you want me to make the change?
Very nice work with the new RequestHandler :+1:

Comment (decebals) [inline pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java]: Sure, you can add a PR that fix this little issue, but without too much noise (with minimal impact) :smiley:

**Pull request merged**